src/main/java/uk/gov/ons/ctp/integration/mockcaseapiservice/CasesConfig.java

```diff
@@ -113,6 +125,36 @@ public synchronized void resetData() throws IOException, CTPException {
     }
   }
 
+  private void checkCaseExists(String strCaseId) throws CTPException {
+    if (!caseUUIDMap.containsKey(strCaseId)) {
+      throw new CTPException(
+          CTPException.Fault.BAD_REQUEST,
+          "Case cannot be modified as UUID not found: " + strCaseId);
+    }
+  }
+
+  private void modifyMaps(CaseContainerDTO caseDetails) {
+    synchronized (caseUUIDMap) {
+      caseUUIDMap.replace(caseDetails.getId().toString(), caseDetails);
+    }
+    synchronized (caseRefMap) {
+      caseRefMap.replace(caseDetails.getCaseRef(), caseDetails);
+    }
+    synchronized (caseUprnMap) {
+      ArrayList<CaseContainerDTO> casesForUprn =
+          (ArrayList<CaseContainerDTO>) caseUprnMap.get(caseDetails.getUprn());
+      for (int i = 0; i < casesForUprn.size(); i++) {
+        CaseContainerDTO caseFound = casesForUprn.get(i);
+        if (caseFound.getId().toString().equals(caseDetails.getId().toString())) {
```

Comment:
I think the .equals() method would consider these two objects to be the same.
I hate iterating like this so I would probs have gone.....

      Set<CaseContainerDTO> caseSet = new HashSet<>();
      caseSet.add(caseDetails);
      caseSet.addAll(caseUprnMap.get(caseDetails.getUprn());
      caseUprnMap.put(caseDetails.getUprn(), new ArrayList<>(caseSet)));

But then I'm lazy....